x/ibc-rate-limit/contracts/rate-limiter/src/packet.rs

```diff
@@ -113,11 +103,15 @@ impl Packet {
     }
 
     pub fn channel_value(&self, deps: Deps, direction: &FlowType) -> Result<Uint256, StdError> {
-        let res = QuerySupplyOfRequest {
-            denom: self.local_denom(direction),
-        }
-        .query(&deps.querier)?;
-        Uint256::from_str(&res.amount.unwrap_or_default().amount)
+        let response: QuerySupplyOfResponse = deps.querier.query(&cosmwasm_std::QueryRequest::Grpc({
+            GrpcQuery {
+                path: "/cosmos.bank.v1beta1.Query/SupplyOf".to_string(),
+                data: Binary::from_base64(
+                    &to_base64(self.local_denom(direction))
+                )?
+            }
+        }))?;
```

Comment:
I think we can reuse the grpc query helpers here just as it is done here https://github.com/neutron-org/neutron-dev-contracts/blob/2589bbdda4496608626d457902d548e36134945a/contracts/grpc_querier/src/contract.rs#L68

Reply:
Makes sense and that would simplify the function a lot, should I remove and change to 

```rust
    pub fn channel_value(&self, deps: Deps, direction: &FlowType) -> Result<Uint256, StdError> {
        Ok(Uint256::from_uint128(deps.querier.query_supply(self.local_denom(direction))?.amount))
    }
```

Reply:
yeah, that should work